[ARM] build/genautomata memory hog

PR target/70473

Refs.: 

 build/genautomata uses 700 MB memory for ARM
  http://gcc.gnu.org/bugzilla/show_bug.cgi?id=60743

 [ARM] Reduce size of arm1020e automaton

	* config/arm/arm1020e.md (1020call_op): Reduce reservation
	duration.
	(v10_fdivs): Likewise.
	(v10_fdivd): Likewise.

--- trunk/gcc/config/arm/arm1020e.md	2016/03/23 10:35:01	234421
+++ trunk/gcc/config/arm/arm1020e.md	2016/03/23 10:39:10	234422
@@ -246,13 +246,14 @@
       (eq_attr "type" "branch"))
  "1020a_e")
 
-;; The latency for a call is not predictable.  Therefore, we use 32 as
-;; roughly equivalent to positive infinity.
+;; The latency for a call is not predictable.  Therefore, we model as blocking
+;; execution for a number of cycles but we can't do anything more accurate
+;; than that.
 
 (define_insn_reservation "1020call_op" 32
  (and (eq_attr "tune" "arm1020e,arm1022e")
       (eq_attr "type" "call"))
- "1020a_e*32")
+ "1020a_e*4")
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; VFP
@@ -300,12 +301,12 @@
 (define_insn_reservation "v10_fdivs" 18
  (and (eq_attr "vfp10" "yes")
       (eq_attr "type" "fdivs, fsqrts"))
- "1020a_e+v10_ds*14")
+ "1020a_e+v10_ds*4")
 
 (define_insn_reservation "v10_fdivd" 32
  (and (eq_attr "vfp10" "yes")
       (eq_attr "type" "fdivd, fsqrtd"))
- "1020a_e+v10_fmac+v10_ds*28")
+ "1020a_e+v10_fmac+v10_ds*4")
 
 (define_insn_reservation "v10_floads" 4
  (and (eq_attr "vfp10" "yes")
-- 
